oShare/Chat Screen/ChatViewController.swift

```diff
@@ -5,35 +5,87 @@ class ChatViewController: UIViewController {
 	
 	// This constraint will be used to adjust the UI in response to the keyboard being displayed/hidden.
 	@IBOutlet private var bottomConstraint: NSLayoutConstraint!
+	@IBOutlet private var textField: UITextField!
+	@IBOutlet private var tableView: UITableView!
 	
 	private var appDelegate: AppDelegate!
+	private var messages: [Dictionary<String, String>] = []
+	private let chatTableViewHandler = ChatTableViewHandler()
 	
 	override func viewDidLoad() {
 		super.viewDidLoad()
 		
 		guard let appDelegate = Constants.appDelegate else { return }
 		self.appDelegate = appDelegate
 		
+		configureTableView()
+		configureTextField()
 		configureEndChatButton()
 		registerKeyboardNotifications()
+		registerDataReceivedNotification()
 	}
 	
 	override func viewWillAppear(_ animated: Bool) {
 		super.viewWillAppear(animated)
 		navigationItem.hidesBackButton = true
 	}
 	
+	private func configureTableView() {
+		chatTableViewHandler.appDelegate = appDelegate
+		chatTableViewHandler.tableView = tableView
+		tableView.delegate = chatTableViewHandler
+		tableView.dataSource = chatTableViewHandler
+	}
+	
+	private func configureTextField() {
+		textField.delegate = self
+		textField.becomeFirstResponder()
+	}
+	
+	// MARK: - Notifications
+	
 	private func registerKeyboardNotifications() {
 		NotificationCenter.default.addObserver(self, selector: #selector(self.adjustForDisplayKeyboard(_:)), name: NSNotification.Name.UIKeyboardWillShow, object: nil)
 		
 		NotificationCenter.default.addObserver(self, selector: #selector(self.adjustForHideKeyboard(_:)), name: NSNotification.Name.UIKeyboardWillHide, object: nil)
 	}
 	
+	private func registerDataReceivedNotification() {
+		NotificationCenter.default.addObserver(self, selector: #selector(self.handleDataReceived(notification:)), name: NSNotification.Name("receivedData"), object: nil)
+	}
+	
+	// MARK: - UI Configuration
+	
 	private func configureEndChatButton() {
 		let endChatButton = UIBarButtonItem(title: "End", style: .done, target: self, action: #selector(self.endChat(sender:)))
 		navigationItem.rightBarButtonItem = endChatButton
 	}
 	
+	// MARK: - Data
+	
+	@objc private func handleDataReceived(notification: Notification) {
+		guard
+			let object = notification.object as? [String: Any],
+			let data = object["data"] as? Data,
+			let peer = object["peer"] as? MCPeerID,
+		  let message = NSKeyedUnarchiver.unarchiveObject(with: data) as? [String: String]
+			else { return }
+		
+		if let message = message["message"] {
+			if message != Constants.Strings.endChatCode {
+				let messageDictionary: [String: String] = ["sender": peer.displayName, "message": message]
+				messages.append(messageDictionary)
+				
+				OperationQueue.main.addOperation { [weak self] () -> Void in
+					guard let strongSelf = self else { return }
+					strongSelf.chatTableViewHandler.reloadData(messages: strongSelf.messages)
+				}
```

Comment:
An explicit route to `main` has been added due to a catch by Main Thread Checker.